packages/dd-trace/src/config.js

```diff
@@ -627,7 +627,11 @@ class Config {
     target.otelMaxQueueSize = nonNegInt(OTEL_BSP_MAX_QUEUE_SIZE, 'OTEL_BSP_MAX_QUEUE_SIZE', false)
 
     const otelMetricsExporter = !OTEL_METRICS_EXPORTER || OTEL_METRICS_EXPORTER.toLowerCase() !== 'none'
-    this.#setBoolean(target, 'otelMetricsEnabled', DD_METRICS_OTEL_ENABLED && otelMetricsExporter)
+    this.#setBoolean(
+      target,
+      'otelMetricsEnabled',
+      DD_METRICS_OTEL_ENABLED && isTrue(DD_METRICS_OTEL_ENABLED) && otelMetricsExporter
```

Comment:
Nit

```suggestion
      otelMetricsExporter && isTrue(DD_METRICS_OTEL_ENABLED)
```

`isTrue()` handles undefined and there is no need to call the method in case the other part is already falsy :)